add an env var for dispatcher debug logging

bdhirsh opened the pull request:
Partially addresses https://github.com/pytorch/pytorch/issues/81703

Committing a patch @zou3519 came up with a while ago - with a `DEBUG=1` build, you can now use `TORCH_SHOW_DISPATCH_TRACE=1 python script.py` to get chatty logs of all of the dispatcher calls that happened in a script.

Stack from [ghstack](https://github.com/ezyang/ghstack):
* #81890
* __->__ #81846

bdhirsh committed:
add an env var for dispatcher debug logging

[ghstack-poisoned]

ezyang commented on aten/src/ATen/core/dispatch/Dispatcher.cpp:
```diff
@@ -4,6 +4,11 @@
 namespace c10 {
 
+bool show_dispatch_trace() {
+    char const* temp = getenv("SHOW_DISPATCH_TRACE");
```
small nits: make this static, and prefix the envvar with TORCH_

bdhirsh committed:
Update on "add an env var for dispatcher debug logging"

Partially addresses https://github.com/pytorch/pytorch/issues/81703

Committing a patch @zou3519 came up with a while ago - with a `DEBUG=1` build, you can now use `SHOW_DISPATCH_TRACE=1 python script.py` to get chatty logs of all of the dispatcher calls that happened in a script.




[ghstack-poisoned]
